pandas/tests/frame/test_combine_concat.py

```diff
@@ -53,6 +53,21 @@ def test_concat_multiple_tzs(self):
         expected = DataFrame(dict(time=[ts2, ts3]))
         assert_frame_equal(results, expected)
 
+    def test_concat_tz_NaT(self):
```

Comment:
Could you also add a test where ts1 is `NaT`?

Reply:
Good catch, it results in an object column instead of the correct dtype. I'm not entirely sure how to solve this issue, as it appears to happen when the dataframe is created before the concat even occurs.

```
ts1 = pd.Timestamp(pd.NaT, tz='UTC')
df1 = pd.DataFrame([[ts1, ts2]])

df1[0]
Out[37]: 
0   NaT
Name: 0, dtype: datetime64[ns]
```

Any ideas?

Reply:
That behavior is theoretically correct if `NaT` was replaced by a naive Timestamp. `NaT` is an edge case where it can exist with a `datetime64[ns, tz]` dtype.

Don't worry about solving that in this PR. Feel free to open up another issue about that case.

Reply:
Sounds good, should I still add the new test?